netaddr.go

```diff
@@ -220,10 +200,152 @@ func parseIPv4(s string) (ip IP, ok bool) {
 		s = s[j:]
 	}
 	if len(s) != 0 {
-		return IP{}, false
+		if s[0] == '%' {
+			return IP{}, parseErr(s, "zone specifier not allowed with IPv4 addresses")
+		}
+		return IP{}, parseErr(s, "trailing garbage after address")
+	}
+
+	return IPv4(ip4[0], ip4[1], ip4[2], ip4[3]), nil
+}
+
+// parseIPv6 parses s as an IPv6 address (in form "2001:db8::68").
+func parseIPv6(s string) (IP, error) {
```

Comment:
Should add a test case for something like `2001:db8:a:b:c:d:e::` or `2001:db8::a:b:c:d:e` and weird edge cases like a single block of zeros being compressed to `::` or trailing segments like that.

Reply:
Done.